src/Typechecker/Monad.hs

```diff
@@ -79,6 +79,11 @@ getPiece = piece <$> ask
 getSize :: Typechecked (Int, Int)
 getSize = size <$> ask
 
+-- | Check whether (x,y) is in the bounds of the board
+inBounds :: (Int, Int) -> Typechecked Bool
+inBounds (x, y) = do
+                    (x', y') <- getSize
+                    return $ x <= x' && y <= y'
```

Comment:
I think this bounds check needs to verify that the x & y components are above 0 here. Not 100% sure, I see it's in place in the other spot, but I imagine it needs to be here too?

Reply:
D'oh! You're right.